apps/src/block_utils.js

```diff
@@ -1010,10 +1010,12 @@ exports.createJsWrapperBlockCreator = function (
     if (eventLoopBlock && args.filter(arg => arg.statement).length === 0) {
       // If the eventloop block doesn't explicitly list its statement inputs,
       // just tack one onto the end
-      args.push({
+      let argsCopy = [...args];
```

Comment:
This was added to prevent the following error:
<img width="803" alt="Screen Shot 2023-10-04 at 7 23 16 PM" src="https://github.com/code-dot-org/code-dot-org/assets/107423305/7bdf7449-dd6b-41d3-a155-0581f1d1e596">
Discussed already with @sanchitmalhotra126 and most likely due to fact that `levelProperties` for Dance lab2 (and not legacy) is stored in Redux.

Reply:
Somewhat tangentially - I'm exploring possibilities around using Blockly's JSON init method, and bypassing these functions in `block_utils`: https://github.com/code-dot-org/code-dot-org/pull/55471. Of course, in order to use this, we'd need to redefine all the dance blocks in Blockly JSON, but we would end up avoiding this code altogether if we do.

Reply:
Oh, but could you add a comment explaining why we need to do this here?